retrofit/src/main/java/retrofit/RestAdapter.java

```diff
@@ -291,7 +291,12 @@ private Object parseResult(MethodInfo methodInfo, Response response)
       }
 
       ResponseBody body = response.body();
-      if (body == null) {
+      if (statusCode == 204 || statusCode == 205) {
+        // HTTP 204 No Content "...response MUST NOT include a message-body"
+        // HTTP 205 Reset Content "...response MUST NOT include an entity"
+        if (body.contentLength() > 0) {
+          throw new IllegalStateException(statusCode + " response must not include body.");
```

Comment:
Not a protocol exception? Somewhat harsh to crash the client when the server made the mistake.